src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -193,9 +193,12 @@ class OrderedRVD(
       return this
     if (shuffle) {
       val shuffled = stably(_.shuffleCoalesce(maxPartitions))
+      val pki = OrderedRVD.getPartitionKeyInfo(typ, OrderedRVD.getKeys(typ, shuffled))
+      if (pki.isEmpty)
```

Comment:
It seems like this check `pki.isEmpty` should be in `OrderedRVD.calculateKeyRanges` since that is where the error happens. Could we either return `OrderedRVD.empty(sparkContext, typ)` in `calculateKeyRanges` or put an assert in `calculateKeyRanges` that `pki` is not empty?

Reply:
added an assert.